packages/rpc-spec/src/rpc-shared.ts

```diff
@@ -0,0 +1,35 @@
+export type RpcRequest<TParams = unknown> = {
+    readonly methodName: string;
+    readonly params: TParams;
+};
+
+export type RpcResponse<TResponse = unknown> = {
+    readonly json: () => Promise<TResponse>;
+    readonly text: () => Promise<string>;
+};
+
+export type RpcRequestTransformer = {
+    <TParams>(request: RpcRequest<unknown>): RpcRequest<TParams>;
+};
+
+export type RpcResponseTransformer = {
+    <TResponse>(response: RpcResponse<unknown>, request: RpcRequest<unknown>): RpcResponse<TResponse>;
```

Comment:
Should `request` be optional? I could imagine someone not caring about the input.

Reply:
It doesn't need to be used by the response transformer but the caller of a response transformer should always provide it, in case it may be needed. Which is way it's not optional in the signature.